simplify `prepare_if_dev` explanation in dbt tutorial (#23765)
dagster-io/dagster: docs/content/integrations/dbt/using-dbt-with-dagster/load-dbt-models.mdx

dbt_project = DbtProject(
    project_dir=Path(__file__).joinpath("..", "..", "..").resolve(),
)

# If this code is imported from `dagster dev`, this generates the manifest.json.
# Otherwise, we expect a manifest to be present in the project's target directory.
dbt_project.prepare_if_dev()
```

Generating the `manifest.json` file for a dbt project is time-consuming, so it's best to avoid doing so every time this Python module is imported. Thus, in production deployments of Dagster, you'll typically have the CI/CD system that packages up your code generate your `manifest.json`.

> However, in development, when you're iteratively making changes to files inside your dbt project and seeing what effects they have, you typically want those changes to be reflected in your local Dagster UI ASAP, without requiring you to explicitly re-generate the manifest. `dbt_project.prepare_if_dev()` helps with this – it re-generates your `manifest.json` at the time Dagster imports your code, _but_ only if it's being imported by the `dagster dev` command.

Reviewer: ```suggestion
However, in development, you typically want changes made to files in your dbt project to be immediately reflected in the Dagster UI without needing to regenerate the manifest. 
`dbt_project.prepare_if_dev()` helps with this – it re-generates your `manifest.json` at the time Dagster imports your code, but only if it's being imported by the `dagster dev` command.
```

I think we can tighten this up more

Author: 👍 good point, made this change
